hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -1004,6 +1007,176 @@ public AbfsRestOperation checkAccess(String path, String rwx, TracingContext tra
+        abfsUriQueryBuilder.toString());
+    final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    final AbfsRestOperation op = new AbfsRestOperation(
+        AbfsRestOperationType.GetBlobProperties,
+        this,
+        HTTP_METHOD_HEAD,
+        url,
+        requestHeaders);
+    op.execute(tracingContext);
+    return op;
+  }
+
+  /**
+   * Call server API <a href="https://learn.microsoft.com/en-us/rest/api/storageservices/list-blobs">BlobList</a>.
+   *
+   * @param sourceDirBlobPath path from where the list of blob is requried.
+   * @param tracingContext object of {@link TracingContext}
+   * @param marker optional value. To be sent in case this method call in a non-first
+   * iteration to the blobList API. Value has to be equal to the field NextMarker in the response
+   * of previous iteration for the same operation.
+   * @param maxResult define how many blobs can client handle in server response.
+   * In case maxResult <= 5000, server sends number of blobs equal to the value. In
+   * case maxResult > 5000, server sends maximum 5000 blobs.
+   * @param absoluteDirSearch
+   *
+   * @return list of {@link BlobProperty}
+   *
+   * @throws AzureBlobFileSystemException thrown from server-call / xml-parsing
+   */
+  public AbfsRestOperation getListBlobs(Path sourceDirBlobPath,
+      TracingContext tracingContext,
+      String marker,
+      String prefix,
+      Integer maxResult, final Boolean absoluteDirSearch)
+      throws AzureBlobFileSystemException {
+    AbfsUriQueryBuilder abfsUriQueryBuilder = createDefaultUriQueryBuilder();
+    abfsUriQueryBuilder.addQuery(QUERY_PARAM_RESTYPE, CONTAINER);
+    abfsUriQueryBuilder.addQuery(QUERY_PARAM_COMP, QUERY_PARAM_COMP_VALUE_LIST);
+    abfsUriQueryBuilder.addQuery(QUERY_PARAM_INCLUDE,
+        QUERY_PARAM_INCLUDE_VALUE_METADATA);
+    if (prefix == null) {
+      prefix = sourceDirBlobPath.toUri().getPath() + (absoluteDirSearch ?"/" : "");
+    }
+    prefix = removeInitialSlash(prefix);
+    abfsUriQueryBuilder.addQuery(QUERY_PARAM_PREFIX, prefix);
+    if (marker != null) {
+      abfsUriQueryBuilder.addQuery(QUERY_PARAM_MARKER, marker);
+    }
+    if (maxResult != null) {
+      abfsUriQueryBuilder.addQuery(QUERY_PARAM_MAXRESULT, maxResult + "");
+    }
+    appendSASTokenToQuery(sourceDirBlobPath.toUri().getPath(),
+        SASTokenProvider.LIST_BLOB_OPERATION, abfsUriQueryBuilder);
+    URL url = createRequestUrl(abfsUriQueryBuilder.toString());
+    final List<AbfsHttpHeader> requestHeaders = createDefaultHeaders();
+    final AbfsRestOperation op = new AbfsRestOperation(
+        AbfsRestOperationType.GetListBlobProperties,
+        this,
+        HTTP_METHOD_GET,
+        url,
+        requestHeaders
+    );
+    op.execute(tracingContext);
+    return op;
+  }
+
+  private String removeInitialSlash(final String prefix) {
+    int len = prefix.length();
+    for (int i = 0; i < len; i++) {
+      if (prefix.charAt(i) != '/') {
```

Comment:
All all places forward and backward slash should be used as a constant,

Reply:
taken.